pkg/fga/client.go

```diff
@@ -379,10 +379,79 @@ type CheckBatchOpts struct {
 	WarrantToken string `json:"-"`
 }
 
+type Warning interface {
+	GetWarning() string
```

Comment:
It's more idiomatic in Go to name this and the other getter `Warning()` and `Code()`. You can then lowercase the `Message` and `Code` attributes in the implementing struct to "unexport" them so the caller must rely on `Warning()` and `Code()`.

On a separate note, why not rename `Warning()` to `Message()`? That way it doesn't stutter (i.e. `Warning.Warning`).

Reply:
Oops yes I meant `Message`